lambda/main.tf

```diff
@@ -55,25 +59,50 @@ resource "aws_kms_ciphertext" "encrypted_environment_variables" {
 }
 
 resource "aws_lambda_event_source_mapping" "sqs_queue_mappings" {
-  for_each                           = var.lambda_sqs_queue_mappings
-  event_source_arn                   = each.value
+  for_each                           = local.sqs_mapping_without_ignore_enabled
+  event_source_arn                   = each.key
   function_name                      = aws_lambda_function.lambda_function.*.arn[0]
   batch_size                         = var.sqs_queue_mapping_batch_size
   maximum_batching_window_in_seconds = var.sqs_queue_batching_window
   dynamic "scaling_config" {
-    for_each = var.sqs_queue_concurrency == null ? [] : [var.sqs_queue_concurrency]
+    for_each = each.value == null ? [] : [each.value]
     content {
       maximum_concurrency = each.value
     }
   }
 }
 
 resource "aws_lambda_permission" "sqs_permissions" {
-  for_each      = var.lambda_sqs_queue_mappings
+  for_each      = local.sqs_mapping_without_ignore_enabled
   action        = "lambda:InvokeFunction"
   function_name = aws_lambda_function.lambda_function.function_name
   principal     = "sqs.amazonaws.com"
-  source_arn    = each.value
+  source_arn    = each.key
+}
```

Comment:
Do we need these resource policies? As far as I'm aware SQS doesn't invoke Lambda, instead a background AWS Lambda process reads from the queue and invokes the function. I can't find any docs that say we need these given the Lambda's role already has permissions `sqs:DeleteMessage`, `sqs:GetQueueAttributes`, and `sqs:ReceiveMessage` on the queue.

Reply:
Ah yeah, you're right, it works without them. I've taken them out.